openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js

```diff
@@ -82,19 +84,44 @@ export default class SelectionManager {
     /**
      * @param {MouseEvent & { currentTarget: HTMLElement }} clickEvent
      */
-    toggleSelected(clickEvent) {
+    processClick(clickEvent) {
         // If there is text selection or the click is on a link that isn't a select handle, don't do anything
-        if (window.getSelection()?.toString() !== '' ||
+        if ((!clickEvent.shiftKey && window.getSelection()?.toString() !== '') ||
             ($(clickEvent.target).closest('a').is('a') &&
             $(clickEvent.target).not('.ile-select-handle').length > 0)) return;
 
         const el = clickEvent.currentTarget;
+        const elIndex = $(el).closest('li,tr')?.index();
+
+        if (clickEvent.shiftKey && this.lastSelectedIndex !== -1 && Math.abs(elIndex - this.lastSelectedIndex) > 1) {
```

Comment:
Hmm we could have a default `lastSelected` here if one is missing ; default to the closest sibling element that's selected. That should handle the case where selection are present on page load, but we can't determine what the 'last' selected element was.

Reply:
This is more complicated than it seems, particularly when you can shift+click either above or below selected elements, and you can shift+click to either select or deselect a range. I think trying to guess the right element to prepopulate here will end up wrong more often than not.